### DISCUSSION TITLE
Can we detect the color scheme in the variant resolver?

### DISCUSSION THREAD

**Discussion by KevinSchiener** [mantinedev/mantine#5029] [Q&A] [answered]
Hello,

we have a custom theme in combination with Mantine 7.x.

We depend on the "light" variant in some components. (MenuItems for example use it by default when setting the color prop.)
Unfortunately we didn't find a way to make light variants work for both, dark and light color scheme, with a shared logic.

In our case, brightening, for example, the font `color` works for dark scheme, but gives us horrible contrast in the light scheme. Darkening the color has the inverse effect.

**Question**: Is there a way to determine the current color scheme in the variant resolver or have a resolver per scheme?

**Comment by rtivital** [comment, marked as answer]
If you do not have ssr in your application, you can get color scheme value from html element: `document.documentElement.getAttribute('data-mantine-color-scheme')`. If you have ssr, then you should customize colors by defining custom CSS variables:

```scss
:root {
  --my-custom-color: light-dark(#fff, #000);
}
```

And then pass these variables to variant resolver.
